feat(Quiz): add quiz component

anderson-olive opened the pull request:
# Description

This PR add the quiz page & components to the project, I've also added the mobile layout version.

## Changes Included:
- Fix mobile layout & add new hamburger navigation
- Add Quiz Page
- Add Quiz Result Page
- Add "QuizItem" Component and "Error" & "Success" variants
- Add "services/quiz" to support retrieve the quiz questions
- UI Improvements

## Demo

Desktop:

https://github.com/nandoramos/youtube-gpt/assets/106548895/64ecbefd-f345-4f61-9cde-9e356b5e10bb

Mobile:

https://github.com/nandoramos/youtube-gpt/assets/106548895/75cd5371-21cc-41ed-93f0-4686a559f4b8

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

anderson-olive committed:
feat(Quiz): add quiz component

- Add Quiz component
- Add mobile layout support
- Quiz basic navigation and workflow

anderson-olive committed:
feat:(QuizResult): display quiz result after quiz

- Display quiz result

anderson-olive committed:
feat(MobileLayout): small layout fix

- Small layout correction
- Close page after select item on menu
- Fix wrong property

nandoramos commented on youtube-gpt-client/src/components/quiz/index.tsx:
```diff
@@ -0,0 +1,128 @@
+    ];
+
+    setQuizStatus(response);
+    moveToNextQuestion(questionIndex, response);
```
Here we shuld add a timeout in order to see the results

nandoramos reviewed (approved):
👏 Very nice!

nnicolopez commented on youtube-gpt-client/src/components/header/index.tsx:
```diff
@@ -1,27 +1,101 @@
+        padding="15px 50px"
+        color="#fff"
+        height="60px"
+        display={['none', 'none', 'flex', 'flex']}
```
Don't get what is this line for. The Chakra-UI Flex component has already `display: flex` as default.

anderson-olive replied:
I don't want to show this component on mobile devices, so we are using the [responsive styles](https://chakra-ui.com/docs/styled-system/responsive-styles) to hide it

nnicolopez commented on youtube-gpt-client/src/components/header/index.tsx:
```diff
@@ -1,27 +1,101 @@
+        color="#fff"
+        height="60px"
+        width="100%"
+        display={['flex', 'flex', 'none', 'none']}
```
Don't get what is this line for. The Chakra-UI Flex component has already display: flex as default.
Same as above.

anderson-olive replied:
Im using [responsive styles](https://chakra-ui.com/docs/styled-system/responsive-styles), using that, we can set the media query property for each media query, and in this case, I want to show it on mobile but not on desktop (smaller screens). I'll simplify this to only 2 media queries (mobile and desktop)

nnicolopez replied:
didn't know that feature, nice!

nnicolopez commented on youtube-gpt-client/src/components/mainContent/index.tsx:
```diff
@@ -9,7 +9,7 @@ const MainContent = ({ children }: PropsWithChildren) => {
       borderRadius="20px 20px 0 0"
       minHeight="calc(100vh - 60px)"
-      padding="55px 50px"
+      padding={['20px 10px', '55px 50px']}
```
```suggestion
      padding="20px 10px 55px 50px"
```

anderson-olive replied:
Same here, we are using different values to each device 20px and 10px to mobile and 55px and 50px to desktop

anderson-olive committed:
feat(Quiz): add timeout to navigation

- Add timeout to the navigation
- Simplify the media query on responsive styles
